webapp/src/client/index.js

```diff
@@ -9,6 +9,10 @@ export const doFetch = async (url, options) => {
     return data;
 };
 
+export const doFetchPromise = (url, options) => {
```

Comment:
What is this for?

Reply:
It's so I can keep the debouncing logic in the component, instead of making a debounced version in the client helper. Of course I'm totally open to a way to improve it.